File: PedidosModel2/WpwtPsmscLogs.cs
#nullable enable
using System;
using System.ComponentModel;
using System.Collections.Generic;

namespace SqlOrganize.Sql.PedidosModel2
{
    public class WpwtPsmscLogs : SqlOrganize.Sql.EntityData
    {

        public override string entityName => "wpwt_psmsc_logs";

        public override void Default()
        {
            EntityVal val = db!.Values("wpwt_psmsc_logs");
            _id = (long?)val.GetDefault("id");
        }


        protected long? _id = null;
        public long? id
        {
            get { return _id; }
            set { _id = value; NotifyPropertyChanged(nameof(id)); }
        }
        protected string? _type = null;
        public string? type
        {
            get { return _type; }
            set { _type = value; NotifyPropertyChanged(nameof(type)); }
        }
        protected long? _ref_id = null;
        public long? ref_id
        {
            get { return _ref_id; }
            set { _ref_id = value; NotifyPropertyChanged(nameof(ref_id)); }
        }
        protected long? _modified_by = null;
        public long? modified_by
        {
            get { return _modified_by; }
            set { _modified_by = value; NotifyPropertyChanged(nameof(modified_by)); }
        }
        protected string? _body = null;
        public string? body
        {
            get { return _body; }
            set { _body = value; NotifyPropertyChanged(nameof(body)); }
        }
        protected DateTime? _date_created = null;
        public DateTime? date_created
        {
            get { return _date_created; }
            set { _date_created = value; NotifyPropertyChanged(nameof(date_created)); }
        }
        protected override string ValidateField(string columnName)
        {

            switch (columnName)
            {

                case "id":
                    if (_id == null)
                        return "Debe completar valor.";
                    return "";

                case "type":
                    if (_type == null)
                        return "Debe completar valor.";
                    return "";

                case "ref_id":
                    if (_ref_id == null)
                        return "Debe completar valor.";
                    return "";

                case "modified_by":
                    if (_modified_by == null)
                        return "Debe completar valor.";
                    return "";

                case "body":
                    if (_body == null)
                        return "Debe completar valor.";
                    return "";

                case "date_created":
                    if (_date_created == null)
                        return "Debe completar valor.";
                    return "";

            }

            return "";
        }
    }
}

File: ModelPedidos/WpwtPsmscLogs.php
<?php

namespace Pedidos;

use SqlOrganize\Sql\Entity;
use Exception;
use DateTime;

class WpwtPsmscLogs extends Entity
{

    public function __construct()
    {
        $this->_entityName = "wpwt_psmsc_logs";
        $this->_db = \SqlOrganize\Sql\DbMy::getInstance();
        $this->setDefault();
    }

    /** @var longtext|null */
    public ?longtext $body = null;

    /** @var DateTime|null */
    public ?DateTime $date_created = null;

    /** @var int|null */
    public ?int $id = null;

    /** @var int|null */
    public ?int $modified_by = null;

    /** @var int|null */
    public ?int $ref_id = null;

    /** @var string|null */
    public ?string $type = null;

    /** @var WpwtPsmscCustomers|null (fk wpwt_psmsc_logs.modified_by _m:o wpwt_psmsc_customers.id) */
    public ?\Pedidos\WpwtPsmscCustomers_ $modified_by_ = null;

}
